src/NUnitEngine/nunit.engine/Services/DefaultTestRunnerFactory.cs

```diff
@@ -76,11 +76,13 @@ public override ITestEngineRunner MakeTestRunner(TestPackage package)
                     projectCount++;
             }
 
+            ProcessModel processModel = GetTargetProcessModel(package);
+
             // If we have multiple projects or a project plus assemblies
             // then defer to the AggregatingTestRunner, which will make
             // the decision on a file by file basis so that each project
             // runs with its own settings.
-            if (projectCount > 1 || projectCount > 0 && assemblyCount > 0)
+            if (projectCount > 1 || (projectCount > 0 && assemblyCount > 0) || (projectCount > 0 && processModel == ProcessModel.Default))
```

Comment:
@CharliePoole option 1 feels cleaner and I do want to take a look at it. Before anything else though I want the integration tests in place which are close to finished. That will give us feedback for this decision. I'm pretty sure option 2 will be the less disruptive option as far as fixing #116 quickly, assuming the integration tests pass this way. A followup PR could then pursue option 1.